### ISSUE TITLE
on my own data

### ISSUE THREAD

**Issue by Hxinyue** [tangjiapeng/SA-ConvONet#11]
Thanks for your work!
 I only have points in .ply format, how can I retrain or generate them directly?

**Comment by tangjiapeng** [comment]
Hi,  you can put your .ply files under  data/demo/synthetic_room_dataset, and then run

python generate_optim_scene.py configs/pointcloud/demo_syn_room.yaml

**Comment by Hxinyue** [comment]
> 

Thank you very much for your reply! Do I need to change my ply format data to the same style as "syntactic_room_dataset"?

**Comment by tangjiapeng** [comment]
You need to normalize the point clouds to [-1, 1].

**Comment by zhao-you-fei** [comment]
您好，请问我在自己的数据集（室内场景）上没有取得很好地效果，可以修改那些参数来增强效果。

**Comment by zhao-you-fei** [comment]
@tangjiapeng
您好，请问我在自己的数据集（室内场景）上没有取得很好地效果，可以修改那些参数来增强效果。

**Comment by Hxinyue** [comment]
> You need to normalize the point clouds to [-1, 1].
Hello, I am trying to run "python generate_optim_scene.py configs/pointcloud/demo_syn_room.yaml" and an error message is reported as follows. May I know where the problem is with me?
![image](https://user-images.githubusercontent.com/108390404/231666883-ee7a26e6-9288-4876-aac1-648519500c95.png)

**Comment by zhao-you-fei** [comment]
非常希望和您讨论学习，我的问题请详见附件




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tangjiapeng/SA-ConvONet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月13日(星期四) 中午1:42
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tangjiapeng/SA-ConvONet] on my own data (Issue #11)





  
You need to normalize the point clouds to [-1, 1].
 Hello, I am trying to run "python generate_optim_scene.py configs/pointcloud/demo_syn_room.yaml" and an error message is reported as follows. May I know where the problem is with me

**Comment by Hxinyue** [comment]
@tangjiapeng Hello, I understand that the loss nature of UCE adds a layer of RulU to the output. What is the relationship between this design and normal line? And whether the "we force the observed surface P to align with the 0.5 level set of occupancy field, and the signed occupancy values of non-surface points to be either 0 or 1"in the article will lose information?

**Comment by tangjiapeng** [comment]
The reason why we don't use common BCE  is that we don't know the inside/outside information due to the lack of normals.
Based on the pre-trained geometric priors, we can still optimize the occupancy fields, by enforcing the occupancy values of non-surface points as either 0 or 1.  The prior initialization can provide some constraints that the points outside the surface would not be pulled to the inside during the optimization of UCE.

**Comment by Hxinyue** [comment]
> The reason why we don't use common BCE is that we don't know the inside/outside information due to the lack of normals. Based on the pre-trained geometric priors, we can still optimize the occupancy fields, by enforcing the occupancy values of non-surface points as either 0 or 1. The prior initialization can provide some constraints that the points outside the surface would not be pulled to the inside during the optimization of UCE.

Hello, may I ask if this zero setting operation will cause loss of output information?

**Comment by zhang-zhe3203** [comment]
您好，非常感谢您的工作，请问我只有一个.ply文件，我想用模型将.ply直接生成mesh并输出，应该怎么做

**Comment by zhang-zhe3203** [comment]
@tangjiapeng作者您好，我尝试使用自己的.ply文件进行mesh生成，我将自己的.ply文件放到data/demo/Matterport3D_processed_normalize路径下，但是缺少同一目录下对应的.npz文件，请问我应该怎么做能得到这个.npz文件。期待您的回复，谢谢！

**Comment by tangjiapeng** [comment]
you can generate the npz file from ply file by "python scripts/dataset_matterport/make_cropscene_dataset2.py --in_folder data/demo/Matterport3D_processed --out_folder data/demo/Matterport3D_processed_normalize --do_norm".

**Comment by zhang-zhe3203** [comment]
> 您可以通过“python scripts/dataset_matterport/make_cropscene_dataset2.py--in_folder data/demo/Materport3D_processed--out_folder-data/demo/Materport 3D_pprocessed_normalize--do_norm”从ply文件生成npz文件。

Thank you very much for your reply. I have successfully run it according to your guidance！

**Comment by Hxinyue** [comment]
Thank you very much for your work, there is no problem at present, it will be closed
